### PR TITLE
cylint: clean the file ternary.pyx

### PR DESCRIPTION
**Pull request by fchapoton**
this is formatting the code in a pyx file (that was not in good shape)

### :memo: Checklist

- [x] The title is concise, informative, and self-explanatory.
- [x] The description explains in detail what this PR is about.
- [ ] I have linked a relevant issue or discussion.
- [ ] I have created tests covering the changes.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: src/sage/quadratic_forms/ternary.pyx [hunk 1]
```diff
@@ -66,182 +66,172 @@ def _reduced_ternary_form_eisenstein_with_matrix(a1, a2, a3, a23, a13, a12):
...
 
-
         # adjust
-        v=a1+a2+a23+a13+a12
-        if (v<0):
-            M*=matrix(ZZ,3,[1,0,1,0,1,1,0,0,1])
-            a3+=v
-            a23+=a12+2*a2
-            a13+=a12+2*a1
+        v = a1+a2+a23+a13+a12
```
Comment (dcoudert) [inline src/sage/quadratic_forms/ternary.pyx:77]:
```diff
-        v = a1+a2+a23+a13+a12
+        v = a1 + a2 + a23 + a13 + a12
```

File: src/sage/quadratic_forms/ternary.pyx [hunk 2]
```diff
@@ -66,182 +66,172 @@ def _reduced_ternary_form_eisenstein_with_matrix(a1, a2, a3, a23, a13, a12):
...
+        if v < 0:
+            M *= matrix(ZZ, 3, [1, 0, 1, 0, 1, 1, 0, 0, 1])
+            a3 += v
+            a23 += a12+2*a2
```
Comment (dcoudert) [inline src/sage/quadratic_forms/ternary.pyx:81]: `a12 + 2*a2` ?

File: src/sage/quadratic_forms/ternary.pyx [hunk 3]
```diff
@@ -66,182 +66,172 @@ def _reduced_ternary_form_eisenstein_with_matrix(a1, a2, a3, a23, a13, a12):
...
-        loop = not (abs(a23) <= a2 and abs(a13) <= a1 and abs(a12) <= a1 and a1+a2+a23+a13+a12>=0)
-
-
-################
-################
-
+                M *= diagonal_matrix([1, 1, -1])
+                a12 = -a12
 
+        loop = not (abs(a23) <= a2 and abs(a13) <= a1 and abs(a12) <= a1 and a1+a2+a23+a13+a12 >= 0)
```
Comment (dcoudert) [inline src/sage/quadratic_forms/ternary.pyx:166]: spaces around `+` ?

File: src/sage/quadratic_forms/ternary.pyx [hunk 4]
```diff
@@ -66,182 +66,172 @@ def _reduced_ternary_form_eisenstein_with_matrix(a1, a2, a3, a23, a13, a12):
...
 
     # adj 3
     if a1+a2+a23+a13+a12 == 0 and 2*a1+2*a13+a12 > 0:
-        M*=matrix(ZZ,3,[-1,0,1,0,-1,1,0,0,1])
+        M *= matrix(ZZ, 3, [-1, 0, 1, 0, -1, 1, 0, 0, 1])
         # a3 += a1+a2+a23+a13+a12
-        a23=-2*a2-a23-a12
-        a13=-2*a1-a13-a12
+        a23 = -2*a2-a23-a12
```
Comment (dcoudert) [inline src/sage/quadratic_forms/ternary.pyx:172]: spaces around `+` and `- ` ?

File: src/sage/quadratic_forms/ternary.pyx [hunk 5]
```diff
@@ -66,182 +66,172 @@ def _reduced_ternary_form_eisenstein_with_matrix(a1, a2, a3, a23, a13, a12):
...
 
     # adj 5.13
     if a1 == -a13 and a12 != 0:
-        M*=matrix(ZZ,3,[-1,0,-1,0,1,0,0,0,-1])
+        M *= matrix(ZZ, 3, [-1, 0, -1, 0, 1, 0, 0, 0, -1])
         # a3 += a1+a13
-        a23=-a23-a12
-        a13=-a13  # = 2*a1+a13
-        a12=-a12
+        a23 = -a23-a12
```
Comment (dcoudert) [inline src/sage/quadratic_forms/ternary.pyx:187]: spaces around `+` and `- ` ?

File: src/sage/quadratic_forms/ternary.pyx [hunk 6]
```diff
@@ -66,182 +66,172 @@ def _reduced_ternary_form_eisenstein_with_matrix(a1, a2, a3, a23, a13, a12):
...
     # adj 5.23
     if a2 == -a23 and a12 != 0:
-        M*=matrix(ZZ,3,[1,0,0,0,-1,-1,0,0,-1])
+        M *= matrix(ZZ, 3, [1, 0, 0, 0, -1, -1, 0, 0, -1])
         # a3 += a2+a23
-        a23=-a23  # = 2*a2+a23
-        a13=-a13-a12
-        a12=-a12
+        a23 = -a23  # = 2*a2+a23
+        a13 = -a13-a12
```
Comment (dcoudert) [inline src/sage/quadratic_forms/ternary.pyx:196]: spaces around `+` and `- ` ?

File: src/sage/quadratic_forms/ternary.pyx [hunk 7]
```diff
@@ -262,124 +252,114 @@ def _reduced_ternary_form_eisenstein_without_matrix(a1, a2, a3, a23, a13, a12):
...
 
     while loop:
 
         # adjust
-        v=a1+a2+a23+a13+a12
-        if (v<0):
-            a3+=v
-            a23+=a12+2*a2
-            a13+=a12+2*a1
+        v = a1+a2+a23+a13+a12
```
Comment (dcoudert) [inline src/sage/quadratic_forms/ternary.pyx:261]: spaces around `+` and `- ` ?

File: src/sage/quadratic_forms/ternary.pyx [hunk 8]
```diff
@@ -262,124 +252,114 @@ def _reduced_ternary_form_eisenstein_without_matrix(a1, a2, a3, a23, a13, a12):
...
-                if (a23==0):
-                    s1=1
-                else:
-                    if (a13==0):
-                        s2=1
-                    else:
-                        if (a12==0):
-                            s3=1
+            [s1, s2, s3] = [a23 > 0, a13 > 0, a12 > 0]
+            if (s1+s2+s3) % 2:
```
Comment (dcoudert) [inline src/sage/quadratic_forms/ternary.pyx:321]: spaces around `+` and `- ` ?

File: src/sage/quadratic_forms/ternary.pyx [hunk 9]
```diff
@@ -262,124 +252,114 @@ def _reduced_ternary_form_eisenstein_without_matrix(a1, a2, a3, a23, a13, a12):
...
-################
-
-
-
     # adj 3
     if a1+a2+a23+a13+a12 == 0 and 2*a1+2*a13+a12 > 0:
         # a3 += a1+a2+a23+a13+a12
-        a23=-2*a2-a23-a12
-        a13=-2*a1-a13-a12
+        a23 = -2*a2-a23-a12
```
Comment (dcoudert) [inline src/sage/quadratic_forms/ternary.pyx:340]: spaces around `+` and `- ` ?

File: src/sage/quadratic_forms/ternary.pyx [hunk 10]
```diff
@@ -470,24 +450,23 @@ def _find_zeros_mod_p_2(a, b, c, r, s, t):
...
         4
-
     """
-
-    zeros=[]
-    v=(1,0,0)
-    if evaluate(a,b,c,r,s,t,v)%2==0:
+    zeros = []
+    v = (1, 0, 0)
+    if evaluate(a, b, c, r, s, t, v) % 2 == 0:
```
Comment (dcoudert) [inline src/sage/quadratic_forms/ternary.pyx:456]: any reason for using `== 0` and not `not ...` ?

File: src/sage/quadratic_forms/ternary.pyx [hunk 11]
```diff
@@ -562,38 +539,37 @@ def _find_zeros_mod_p_odd(long long a, long long b, long long c, long long r, lo
     y0 = v[1]
     more = False
     for i in range(p):
-        a_i=(a*x0**2+b*i**2-2*b*i*y0+b*y0**2-t*i*x0+t*x0*y0-2*a*x0+t*i-t*y0+s*x0-r*i+r*y0+a+c-s)%p
-        #b_i=(-2*b*i**2+2*b*i*y0+t*i*x0+2*a*x0-2*t*i+t*y0+r*i-2*a+s)%p
-        if a_i==0:
-            w=((x0-1)%p,(y0-i)%p,1)
-            if w==v:
-                more=True
+        a_i = (a*x0**2+b*i**2-2*b*i*y0+b*y0**2-t*i*x0+t*x0*y0-2*a*x0+t*i-t*y0+s*x0-r*i+r*y0+a+c-s) % p
```
Comment (dcoudert) [inline src/sage/quadratic_forms/ternary.pyx:542]: spaces around `+` and `- ` ?

File: src/sage/quadratic_forms/ternary.pyx [hunk 12]
```diff
@@ -562,38 +539,37 @@ def _find_zeros_mod_p_odd(long long a, long long b, long long c, long long r, lo
...
+            if w[0] == v[0] and w[1] == v[1] and w[2] == v[2]:
+                more = True
             else:
                 zeros.append(w)
     if more:
-        a_inf=(a*x0**2+b*y0**2+t*x0*y0-2*b*y0-t*x0+s*x0+r*y0+b+c-r)%p
-        #b_inf=(2*b*y0+t*x0-2*b+r)%p
-        if a_inf==0:
-            w=(x0%p,(y0-1)%p,1)
+        a_inf = (a*x0**2+b*y0**2+t*x0*y0-2*b*y0-t*x0+s*x0+r*y0+b+c-r) % p
```
Comment (dcoudert) [inline src/sage/quadratic_forms/ternary.pyx:559]: spaces around `+` and `- ` ?

File: src/sage/quadratic_forms/ternary.pyx [hunk 13]
```diff
@@ -672,54 +641,45 @@ def _find_all_ternary_qf_by_level_disc(long long N, long long d):
...
+                    if s*t == 0:
+                        r_max = 0
                     else:
-                        r_max=b
-                    while r<=r_max:
+                        r_max = b
+                    while r <= r_max:
 
-                        if (d-r*s*t+a*r*r+b*s*s)%(4*a*b-t**2)==0:
+                        if (d-r*s*t+a*r*r+b*s*s) % (4*a*b-t**2) == 0:
```
Comment (dcoudert) [inline src/sage/quadratic_forms/ternary.pyx:680]: spaces around `+` and `- ` ?

**Review by dcoudert — CHANGES_REQUESTED** [review]
a few comments.

File: src/sage/quadratic_forms/ternary.pyx (hunk 13 above)
Comment (fchapoton) [reply src/sage/quadratic_forms/ternary.pyx:680]: salut David,
ajouter des espaces autour des operations arithmetiques (`+, *, -, /`) est une tache herculeenne, que je prefere laisser de coté pour le moment. Il y a environ 40000 messages pour l'erreur pycodestyle E226 dans les fichiers python de sage, et idem dans les fichiers cython

Comment (dcoudert) [reply src/sage/quadratic_forms/ternary.pyx:680]: Je comprends. Je trouvais étrange que tu ne le fasses pas en même temps sur les lignes que tu as modifié. On va donc laisser comme ça.

**Review by dcoudert — APPROVED** [review]
LGTM.

**Pull request merged**